lib/package-snippets-view.js

```diff
@@ -110,19 +150,86 @@ export default class PackageSnippetsView {
         nameTd.textContent = name
         row.appendChild(nameTd)
 
+        const scopeTd = document.createElement('td')
+        scopeTd.classList.add('snippet-scope-name')
+        scopeTd.textContent = selectorString
+        row.appendChild(scopeTd)
+
         const bodyTd = document.createElement('td')
         bodyTd.classList.add('snippet-body')
-        bodyTd.textContent = body
         row.appendChild(bodyTd)
 
         this.refs.snippets.appendChild(row)
+        this.createButtonsForSnippetRow(bodyTd, {body, prefix, scope: selectorString, name})
       }
 
       if (this.refs.snippets.children.length > 0) {
-        this.refs.snippets.style.display = ''
+        this.element.style.display = ''
       } else {
-        this.refs.snippets.style.display = 'none'
+        this.element.style.display = 'none'
       }
     })
   }
+
+  createButtonsForSnippetRow (td, {scope, body, name, prefix}) {
+    let buttonContainer = document.createElement('div')
+    buttonContainer.classList.add('btn-group', 'btn-group-xs')
+
+    let viewButton = document.createElement('button')
+    let copyButton = document.createElement('button')
+
+    viewButton.setAttribute('type', 'button')
+    viewButton.textContent = 'View'
+    viewButton.classList.add('btn', 'snippet-view-btn')
+
+    let tooltip = atom.tooltips.add(viewButton, {
+      title: body,
+      html: false,
+      trigger: 'click',
+      placement: 'auto left',
+      'class': 'snippet-body-tooltip'
+    })
+
+    this.disposables.add(tooltip)
+
+    copyButton.setAttribute('type', 'button')
+    copyButton.textContent = 'Copy'
+    copyButton.classList.add('btn', 'snippet-copy-btn')
+
+    copyButton.addEventListener('click', (event) => {
+      event.preventDefault()
+      return this.writeSnippetToClipboard({scope, body, name, prefix})
+    })
+
+    buttonContainer.appendChild(viewButton)
+    buttonContainer.appendChild(copyButton)
+
+    td.appendChild(buttonContainer)
+  }
+
+  writeSnippetToClipboard ({scope, body, name, prefix}) {
+    let content
+    const extension = path.extname(this.snippetsProvider.getUserSnippetsPath())
+    if (extension === '.cson') {
+      body = body.replace(/'/g, `\\'`).replace(/\n/g, '\\n')
+      content = `
+'${scope}':
+  '${name}':
+    'prefix': '${prefix}'
+    'body': '${body}'
+`
+    } else {
+      body = body.replace(/"/g, `\\"`).replace(/\n/g, '\\n')
```

Comment:
Oops. I completely forgot that tabs should be replaced with `\t` when writing my review comment about newlines. The article snippet copies the newlines correctly but the tab character is still there.